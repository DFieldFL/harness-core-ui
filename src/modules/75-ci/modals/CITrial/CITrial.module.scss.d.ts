/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly ciTrialSelect: string
}
export default styles
