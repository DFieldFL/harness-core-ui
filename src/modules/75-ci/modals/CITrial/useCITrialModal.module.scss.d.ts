/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly dialog: string
  readonly ciTrial: string
}
export default styles
