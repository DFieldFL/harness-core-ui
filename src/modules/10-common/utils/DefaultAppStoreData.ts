import type { AppStoreContextProps } from 'framework/AppStore/AppStoreContext'
import type { Project } from 'services/cd-ng'

export const project: Project = {
  orgIdentifier: 'testOrg',
  identifier: 'test',
  name: 'test',
  color: '#e6b800',
  modules: ['CD', 'CV'],
  description: 'test',
  tags: { tag1: '', tag2: 'tag3' }
}
export const defaultAppStoreValues: Pick<
  AppStoreContextProps,
  'selectedProject' | 'featureFlags' | 'currentUserInfo'
> = {
  selectedProject: {
    orgIdentifier: 'testOrg',
    identifier: 'test',
    name: 'test',
    color: '#e6b800',
    modules: ['CD', 'CV', 'CI', 'CE', 'CF'],
    description: 'test',
    tags: { tag1: '', tag2: 'tag3' }
  },
  featureFlags: {
    CDNG_ENABLED: true,
    CVNG_ENABLED: true,
    CING_ENABLED: true,
    CENG_ENABLED: true,
    CFNG_ENABLED: true
  },
  currentUserInfo: {
    uuid: 'dummyId',
    name: 'dummyname',
    email: 'dummy@harness.io',
    admin: false,
    twoFactorAuthenticationEnabled: false
  }
}
