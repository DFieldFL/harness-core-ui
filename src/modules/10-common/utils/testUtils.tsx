import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uikit'
import qs from 'qs'

import strings from 'strings/strings.en.yaml'
import { AppStoreProvider, useAppStoreWriter } from 'framework/hooks/useAppStore'
import type { AppStore } from 'framework/types/AppStore'
import { AUTH_ROUTE_PATH_PREFIX, StringsContext } from 'framework/exports'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): Element | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): Element | null => document.querySelector('.bp3-popover-content')

interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, any>
  defaultAppStoreValues?: Partial<AppStore>
}

export const prependAccountPath = (path: string): string => AUTH_ROUTE_PATH_PREFIX + path

const AppStoreInitializer: React.FC<{ defaultAppStoreValues: TestWrapperProps['defaultAppStoreValues'] }> = ({
  children,
  defaultAppStoreValues
}) => {
  const updateAppStore = useAppStoreWriter()
  updateAppStore(defaultAppStoreValues || {})
  return <> {children} </>
}

export const NotFound = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  const { path = '/', pathParams = {}, defaultAppStoreValues, queryParams = {} } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  const history = createMemoryHistory({ initialEntries: [routePath] })

  return (
    <AppStoreProvider>
      <StringsContext.Provider value={strings}>
        <AppStoreInitializer defaultAppStoreValues={defaultAppStoreValues}>
          <Router history={history}>
            <ModalProvider>
              <RestfulProvider base="/">
                <Switch>
                  <Route exact path={path}>
                    {props.children}
                  </Route>
                  <Route>
                    <NotFound />
                  </Route>
                </Switch>
              </RestfulProvider>
            </ModalProvider>
          </Router>
        </AppStoreInitializer>
      </StringsContext.Provider>
    </AppStoreProvider>
  )
}
