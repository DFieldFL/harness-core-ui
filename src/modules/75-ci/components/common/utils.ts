import { parse } from 'yaml'
import { get } from 'immer/dist/internal'
import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
import { loggerFor } from 'framework/logging/logging'
import type { CompletionItemInterface } from '@common/interfaces/YAMLBuilderProps'
import { ModuleName } from 'framework/types/ModuleName'
import { getConnectorSuggestions } from '../PipelineSteps/EditorSuggestionUtils'

const logger = loggerFor(ModuleName.CI)

/* istanbul ignore next */
export const getConnectorList = async (
  connectorTypes: string[] | undefined,
  path: string,
  yaml: string,
  params: Record<string, unknown>
): Promise<CompletionItemInterface[]> => {
  if (connectorTypes?.length) {
    return getConnectorSuggestions(params, connectorTypes)
  }
  let pipelineObj
  try {
    pipelineObj = parse(yaml)
  } catch (err) {
    logger.error('Error while parsing the yaml', err)
  }
  if (pipelineObj) {
    const obj = get(pipelineObj, path.replace('.spec.connectorRef', ''))
    if (obj.type === StepType.Dependency) {
      return getConnectorSuggestions(params, ['Gcp', 'Aws', 'DockerRegistry'])
    }
  }
  return []
}
