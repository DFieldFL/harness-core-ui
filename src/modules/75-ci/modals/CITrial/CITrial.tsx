import React, { useState } from 'react'
import { Text, Layout, Icon, Container, Color } from '@wings-software/uicore'
import { useStrings } from 'framework/exports'
import type { NgPipeline } from 'services/cd-ng'
import { SelectOrCreatePipelineForm } from '@pipeline/components/SelectOrCreatePipelineForm/SelectOrCreatePipelineForm'
import { CreatePipelineForm } from '@pipeline/components/CreatePipelineForm/CreatePipelineForm'
import ciImage from '../images/illustration.png'

type handleSelectSubmit = (value: string) => void
type handleCreateSubmit = (value: NgPipeline) => void
interface CITrialModalData {
  onSubmit: handleSelectSubmit | handleCreateSubmit
  closeModal?: () => void
  isSelect: boolean
}

const CITrial: React.FC<CITrialModalData> = ({ isSelect, onSubmit, closeModal }) => {
  const { getString } = useStrings()

  const [select, setSelect] = useState(isSelect)

  return (
    <Layout.Vertical padding={{ top: 'large', left: 'xxxlarge' }}>
      <Layout.Horizontal padding={{ top: 'large' }} spacing="small">
        <Icon name="ci-main" size={20} />
        <Text style={{ color: Color.BLACK, fontSize: 'medium' }}>{getString('ci.continuous')}</Text>
        <Text
          style={{
            backgroundColor: 'var(--orange-500)',
            color: Color.WHITE,
            textAlign: 'center',
            width: 120,
            borderRadius: 3,
            marginLeft: 30,
            marginTop: 6,
            display: 'inline-block'
          }}
        >
          {getString('common.trialInProgress')}
        </Text>
      </Layout.Horizontal>
      <Layout.Horizontal padding={{ top: 'large' }}>
        <Container width="57%" padding={{ right: 'xxxlarge' }}>
          <Text style={{ fontSize: 'normal', width: 380, display: 'inline-block', marginLeft: 30, lineHeight: 2 }}>
            {select
              ? getString('pipeline.selectOrCreateForm.description')
              : getString('ci.ciTrialHomePage.startTrial.description')}
          </Text>
          <img src={ciImage} style={{ marginLeft: -40, marginTop: -30 }} width={800} height={400} />
        </Container>
        <Container width="30%" padding={{ left: 'xxxlarge' }} border={{ left: true }} height={400}>
          {select ? (
            <SelectOrCreatePipelineForm
              handleSubmit={onSubmit as handleSelectSubmit}
              openCreatPipeLineModal={() => {
                setSelect(false)
              }}
              closeModal={closeModal}
            />
          ) : (
            <CreatePipelineForm handleSubmit={onSubmit as handleCreateSubmit} closeModal={closeModal} />
          )}
        </Container>
      </Layout.Horizontal>
    </Layout.Vertical>
  )
}

export default CITrial
