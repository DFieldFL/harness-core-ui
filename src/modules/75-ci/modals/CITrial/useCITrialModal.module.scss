.dialog {
  border-radius: 11px;
  position: relative;
  padding: unset !important;
  &.ciTrial {
    width: 1000px;
    border-left: none !important;
  }
}
