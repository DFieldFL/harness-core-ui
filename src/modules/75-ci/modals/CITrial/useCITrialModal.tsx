import React from 'react'
import { useModalHook, Button } from '@wings-software/uicore'
import { Dialog, Classes } from '@blueprintjs/core'
import cx from 'classnames'
import { useStrings } from 'framework/strings'
import { useTelemetry } from '@common/hooks/useTelemetry'
import { Category, TrialActions, PageNames } from '@common/constants/TrackingConstants'

import { TrialModalTemplate } from '@templates-library/components/TrialModalTemplate/TrialModalTemplate'
import type {
  UseTrialModalProps,
  PipelineProps,
  UseTrialModalReturns
} from '@templates-library/components/TrialModalTemplate/trialModalUtils'
import { useGetFormPropsByTrialType } from '@templates-library/components/TrialModalTemplate/trialModalUtils'
import ciImage from '../images/illustration.png'

import css from './useCITrialModal.module.scss'

const CITrial: React.FC<UseTrialModalProps> = ({ trialType, actionProps }) => {
  const { getString } = useStrings()

  const { child, description, rightWidth } = useGetFormPropsByTrialType({ trialType, actionProps, module: 'ci' })

  return (
    <TrialModalTemplate
      iconName="ci-main"
      title={getString('ci.continuous')}
      description={description}
      imgSrc={ciImage}
      rightWidth={rightWidth}
    >
      {child}
    </TrialModalTemplate>
  )
}

const CITrialDialog = ({ actionProps, trialType }: UseTrialModalProps): React.ReactElement => {
  const { trackEvent } = useTelemetry()
  const handleClose = (): void => {
    const pipelineProps = actionProps as PipelineProps
    pipelineProps.onCloseModal?.()
    trackEvent(TrialActions.TrialModalPipelineSetupCancel, { category: Category.SIGNUP, module: 'ci' })
  }

  useTelemetry({
    pageName: PageNames.TrialSetupPipelineModal,
    category: Category.SIGNUP,
    properties: { module: 'ci' }
  })

  return (
    <Dialog
      isOpen={true}
      enforceFocus={false}
      onClose={handleClose}
      className={cx(css.dialog, Classes.DIALOG, css.ciTrial)}
    >
      <CITrial trialType={trialType} actionProps={actionProps} />
      <Button
        aria-label="close modal"
        minimal
        icon="cross"
        iconProps={{ size: 18 }}
        onClick={handleClose}
        className={css.crossIcon}
      />
    </Dialog>
  )
}

export const getCITrialDialog = ({ actionProps, trialType }: UseTrialModalProps): React.ReactElement => (
  <CITrialDialog actionProps={actionProps} trialType={trialType} />
)

export const useCITrialModal = ({ actionProps, trialType }: UseTrialModalProps): UseTrialModalReturns => {
  const [showModal, hideModal] = useModalHook(() => {
    const onCloseModal = (): void => {
      const pipelineProps = actionProps as PipelineProps
      hideModal()
      pipelineProps.onCloseModal?.()
    }
    const newActionProps = { ...actionProps, onCloseModal }
    return <CITrialDialog actionProps={newActionProps} trialType={trialType} />
  })

  return {
    openTrialModal: showModal,
    closeTrialModal: hideModal
  }
}
