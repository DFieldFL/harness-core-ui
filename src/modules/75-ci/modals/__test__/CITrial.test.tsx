import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { defaultAppStoreValues } from '@common/utils/DefaultAppStoreData'

import CITrial from '../CITrial/CITrial'

const reloadPipelinesMock = (): Promise<{ status: string }> => {
  return Promise.resolve({ status: 'SUCCESS' })
}

jest.mock('services/pipeline-ng', () => ({
  useGetPipelineList: jest.fn().mockImplementation(() => {
    return {
      cancel: jest.fn(),
      loading: false,
      mutate: reloadPipelinesMock
    }
  })
}))

describe('CITrial Modal', () => {
  const props = {
    onSubmit: jest.fn(),
    isSelect: false
  }
  describe('Rendering', () => {
    test('should render CITrial with Create Pipeline Form when isSelect is false', () => {
      const { container, getByText } = render(
        <TestWrapper
          path="/account/:accountId"
          pathParams={{ accountId: 'testAcc' }}
          defaultAppStoreValues={defaultAppStoreValues}
        >
          <CITrial {...props} />
        </TestWrapper>
      )
      expect(getByText('ci.ciTrialHomePage.startTrial.description')).toBeDefined()
      expect(container).toMatchSnapshot()
    })

    test('should render CITrial with Select Pipeline Form when isSelect is true', () => {
      const { container, getByText } = render(
        <TestWrapper
          path="/account/:accountId"
          pathParams={{ accountId: 'testAcc' }}
          defaultAppStoreValues={defaultAppStoreValues}
        >
          <CITrial {...props} isSelect={true} />
        </TestWrapper>
      )
      expect(getByText('pipeline.selectOrCreateForm.description')).toBeDefined()
      expect(container).toMatchSnapshot()
    })
  })
})
