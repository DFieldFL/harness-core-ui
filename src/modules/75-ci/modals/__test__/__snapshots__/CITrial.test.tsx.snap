// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CITrial Modal Rendering should render CITrial with Create Pipeline Form when isSelect is false 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main StyledProps--padding-top-large StyledProps--padding-left-xxxlarge"
  >
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--padding-top-large"
    >
      <span
        class="bp3-icon StyledProps--main"
        data-icon="ci-main"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <path
            clip-rule="evenodd"
            d="M12.005 10.97a2.9 2.9 0 01-3.003-2.96 2.9 2.9 0 013.002-3.008 2.927 2.927 0 012.994 3.007 2.882 2.882 0 01-2.993 2.989v-.027zM6 .9a9.987 9.987 0 014.203-.9C15.56-.053 19.945 4.258 20 9.63v.144A9.928 9.928 0 0119.103 14l-3.092-3.102a5.15 5.15 0 00.762-2.697 4.743 4.743 0 00-4.93-4.937 5.013 5.013 0 00-2.688.765L6 .899zM3.991 2.2l-.099.09L7.36 5.734a5.16 5.16 0 00-.766 2.762c-.081 2.635 1.977 4.838 4.597 4.921.116.003.232.002.349-.002a5.116 5.116 0 002.774-.77L17.8 16.13A9.792 9.792 0 019.91 20c-5.393.08-9.829-4.25-9.908-9.673L0 10.153A9.645 9.645 0 013.99 2.2z"
            fill="url(#ci-main_svg__paint0_linear)"
            fill-rule="evenodd"
          />
          <defs>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="ci-main_svg__paint0_linear"
              x1="-6.46"
              x2="6.46"
              y1="6.46"
              y2="26.46"
            >
              <stop
                stop-color="#73DFE7"
              />
              <stop
                offset="1"
                stop-color="#0095F7"
              />
            </lineargradient>
          </defs>
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main"
        style="color: black; font-size: medium;"
      >
        ci.continuous
      </p>
      <p
        class="StyledProps--font StyledProps--main"
        style="color: white; text-align: center; width: 120px; border-radius: 3px; margin-left: 30px; display: inline-block;"
      >
        common.trialInProgress
      </p>
    </div>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--padding-top-large"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-right-xxxlarge"
        style="width: 57%;"
      >
        <p
          class="StyledProps--font StyledProps--main"
          style="width: 380px; display: inline-block; margin-left: 30px; line-height: 2;"
        >
          ci.ciTrialHomePage.startTrial.description
        </p>
        <img
          height="400"
          src="test-file-stub"
          style="margin-left: -40px; margin-top: -30px;"
          width="800"
        />
      </div>
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-left-xxxlarge StyledProps--border StyledProps--border-left"
        style="width: 30%; height: 400px;"
      >
        <div
          class="OverlaySpinner--overlaySpinner"
        >
          <form
            action="#"
            class="FormikForm--main"
          >
            <div>
              <p
                class="StyledProps--font StyledProps--main"
                style="color: black; padding-bottom: 8px; font-weight: 600; font-size: large;"
              >
                pipeline.createPipeline.setupHeader
              </p>
              <p
                class="StyledProps--font StyledProps--main"
                style="color: black; padding-bottom: 40px;"
              >
                pipeline.createPipeline.setupSubtitle
              </p>
              <div
                class="StyledProps--font StyledProps--main main"
              >
                <div
                  class="InputWithIdentifier--txtNameContainer TextInput--main"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
                  >
                    <p
                      class="StyledProps--font StyledProps--main"
                    >
                      ID
                      :
                    </p>
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <div
                          class="bp3-editable-text bp3-editable-text-placeholder InputWithIdentifier--idValue"
                          tabindex="0"
                        >
                          <span
                            class="bp3-editable-text-content"
                            style="height: 1px; line-height: 1px; min-width: 0;"
                          />
                        </div>
                      </span>
                    </span>
                    <span
                      class="bp3-icon bp3-icon-edit StyledProps--main"
                      icon="edit"
                      style="vertical-align: middle;"
                    >
                      <svg
                        data-icon="edit"
                        height="12"
                        viewBox="0 0 16 16"
                        width="12"
                      >
                        <desc>
                          edit
                        </desc>
                        <path
                          d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="bp3-form-group"
                    data-id="name-0"
                  >
                    <label
                      class="bp3-label"
                      for="name"
                    >
                      name
                       
                      <span
                        class="bp3-text-muted"
                      />
                    </label>
                    <div
                      class="bp3-form-content"
                    >
                      <div
                        class="bp3-input-group"
                      >
                        <input
                          autocomplete="off"
                          class="bp3-input"
                          name="name"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="StyledProps--font StyledProps--main"
                >
                  <label
                    class="Label--main"
                    style="font-size: 13px; font-weight: normal;"
                  >
                    optionalField
                    <span
                      class="bp3-icon bp3-icon-edit StyledProps--main editOpen StyledProps--color StyledProps--color-grey500"
                      data-name="edit"
                      data-testid="description-edit"
                      icon="edit"
                    >
                      <svg
                        data-icon="edit"
                        height="12"
                        viewBox="0 0 16 16"
                        width="12"
                      >
                        <desc>
                          edit
                        </desc>
                        <path
                          d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </label>
                </div>
                <div
                  class="StyledProps--font StyledProps--main"
                >
                  <label
                    class="Label--main"
                    style="font-size: 13px; font-weight: normal;"
                  >
                    optionalField
                    <span
                      class="bp3-icon bp3-icon-edit StyledProps--main editOpen StyledProps--color StyledProps--color-grey500"
                      data-name="edit"
                      data-testid="tags-edit"
                      icon="edit"
                    >
                      <svg
                        data-icon="edit"
                        height="12"
                        viewBox="0 0 16 16"
                        width="12"
                      >
                        <desc>
                          edit
                        </desc>
                        <path
                          d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--padding-top-large"
              >
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
                  type="submit"
                >
                  <span
                    class="bp3-button-text"
                  >
                    start
                  </span>
                </button>
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color"
                  type="reset"
                >
                  <span
                    class="bp3-button-text"
                  >
                    pipeline.createPipeline.setupLater
                  </span>
                </button>
              </div>
              <div
                class="Layout--horizontal StyledProps--main StyledProps--padding-top-large"
              >
                <a
                  href="/"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                  >
                    <p
                      class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-blue700 StyledProps--font-normal"
                      style="display: flex; align-items: center;"
                    >
                      pipeline.createPipeline.learnMore
                      <span
                        class="bp3-icon bp3-icon-chevron-right StyledProps--main StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-blue700"
                        icon="chevron-right"
                      >
                        <svg
                          data-icon="chevron-right"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            chevron-right
                          </desc>
                          <path
                            d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </p>
                  </div>
                </a>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CITrial Modal Rendering should render CITrial with Select Pipeline Form when isSelect is true 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main StyledProps--padding-top-large StyledProps--padding-left-xxxlarge"
  >
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--padding-top-large"
    >
      <span
        class="bp3-icon StyledProps--main"
        data-icon="ci-main"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <path
            clip-rule="evenodd"
            d="M12.005 10.97a2.9 2.9 0 01-3.003-2.96 2.9 2.9 0 013.002-3.008 2.927 2.927 0 012.994 3.007 2.882 2.882 0 01-2.993 2.989v-.027zM6 .9a9.987 9.987 0 014.203-.9C15.56-.053 19.945 4.258 20 9.63v.144A9.928 9.928 0 0119.103 14l-3.092-3.102a5.15 5.15 0 00.762-2.697 4.743 4.743 0 00-4.93-4.937 5.013 5.013 0 00-2.688.765L6 .899zM3.991 2.2l-.099.09L7.36 5.734a5.16 5.16 0 00-.766 2.762c-.081 2.635 1.977 4.838 4.597 4.921.116.003.232.002.349-.002a5.116 5.116 0 002.774-.77L17.8 16.13A9.792 9.792 0 019.91 20c-5.393.08-9.829-4.25-9.908-9.673L0 10.153A9.645 9.645 0 013.99 2.2z"
            fill="url(#ci-main_svg__paint0_linear)"
            fill-rule="evenodd"
          />
          <defs>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="ci-main_svg__paint0_linear"
              x1="-6.46"
              x2="6.46"
              y1="6.46"
              y2="26.46"
            >
              <stop
                stop-color="#73DFE7"
              />
              <stop
                offset="1"
                stop-color="#0095F7"
              />
            </lineargradient>
          </defs>
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main"
        style="color: black; font-size: medium;"
      >
        ci.continuous
      </p>
      <p
        class="StyledProps--font StyledProps--main"
        style="color: white; text-align: center; width: 120px; border-radius: 3px; margin-left: 30px; display: inline-block;"
      >
        common.trialInProgress
      </p>
    </div>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--padding-top-large"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-right-xxxlarge"
        style="width: 57%;"
      >
        <p
          class="StyledProps--font StyledProps--main"
          style="width: 380px; display: inline-block; margin-left: 30px; line-height: 2;"
        >
          pipeline.selectOrCreateForm.description
        </p>
        <img
          height="400"
          src="test-file-stub"
          style="margin-left: -40px; margin-top: -30px;"
          width="800"
        />
      </div>
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-left-xxxlarge StyledProps--border StyledProps--border-left"
        style="width: 30%; height: 400px;"
      >
        <div
          class="OverlaySpinner--overlaySpinner"
        >
          <form
            action="#"
            class="FormikForm--main"
          >
            <div>
              <p
                class="StyledProps--font StyledProps--main"
                style="color: black; padding-bottom: 8px; font-weight: 600; font-size: large;"
              >
                pipeline.selectOrCreatePipeline.setupHeader
              </p>
              <p
                class="StyledProps--font StyledProps--main"
                style="color: black; padding-bottom: 40px;"
              >
                common.letsGetYouStarted
              </p>
              <div
                class="Layout--vertical Layout--layout-spacing-small StyledProps--main StyledProps--padding-top-large StyledProps--padding-bottom-small"
              >
                <p
                  class="StyledProps--font StyledProps--main"
                  style="color: black;"
                >
                  pipeline.selectOrCreatePipeline.selectAPipeline
                </p>
                <div
                  class="bp3-popover-wrapper select"
                >
                  <div
                    class="bp3-popover-target"
                  >
                    <div
                      class=""
                    >
                      <button
                        class="bp3-button main"
                        data-testid="pipeline-select"
                        type="button"
                      >
                        <span
                          class="bp3-button-text"
                        >
                           
                        </span>
                        <span
                          class="bp3-icon bp3-icon-chevron-down"
                          icon="chevron-down"
                        >
                          <svg
                            data-icon="chevron-down"
                            height="16"
                            viewBox="0 0 16 16"
                            width="16"
                          >
                            <desc>
                              chevron-down
                            </desc>
                            <path
                              d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="bp3-form-group pipelineRequired"
                  data-id="pipelineRequired-0"
                >
                  <div
                    class="bp3-form-content"
                  >
                    <div
                      class="bp3-input-group"
                    >
                      <input
                        autocomplete="off"
                        class="bp3-input"
                        name="pipelineRequired"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--padding-top-large"
              >
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
                  type="submit"
                >
                  <span
                    class="bp3-button-text"
                  >
                    continue
                  </span>
                </button>
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color"
                  style="text-align: center; padding: 7px; width: 150px; height: 33px; display: inline-block; border-width: 1px;"
                  type="button"
                >
                  <span
                    class="bp3-button-text"
                  >
                    pipeline.createANewPipeline
                  </span>
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
